pravega-client-examples/src/main/java/io/pravega/example/secure/SecureReader.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2019 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ */
+package io.pravega.example.secure;
+
+import io.pravega.client.ClientConfig;
+import io.pravega.client.ClientFactory;
+import io.pravega.client.admin.ReaderGroupManager;
+import io.pravega.client.stream.*;
+import io.pravega.client.stream.impl.DefaultCredentials;
+import io.pravega.client.stream.impl.JavaSerializer;
+
+/**
+ * This class demonstrates how to configure a Pravega reader application for:
+ *
+ *    a) SSL/TLS (HTTPS) communications with a Pravega cluster for data-in-transit encryption and
+ *       server authentication.
+ *    b) Passing credentials to a Pravega cluster for authentication and authorization of clients.
+ *
+ * This example can be driven interactively against a running Pravega cluster configured to communicate using SSL/TLS
+ * and "auth" (authentication and authorization) turned on.
+ */
+public class SecureReader {
+
+    public static void main(String[] args) throws ReinitializationRequiredException {
+
+        /**
+         * Note about setting the client config for HTTPS:
+         *    - The client config below is configured to use an optional truststore. The truststore is expected to be
+         *      the certificate of the certification authority (CA) that was used to sign the server certificates.
+         *      If this is null or empty, the default JVM trust store is used. In this demo, we use a provided
+         *      "cert.pem" as the CA certificate, which is also provided on the server-side. If the cluster uses a
+         *      different CA (which it should), use that CA's certificate as the truststore instead.
+         *
+         *    - Also, the client config below disables host name verification. If the cluster's server certificates
+         *      have host names specified as the that of the server, you may turn this on. In a production
+         *      deployment, it is recommended to keep this on.
+         *
+         * Note about setting the client config for auth:
+         *    - The client config below is configured with an object of DefaultCredentials class. The user name
+         *      and password arguments passed to the object represent the credentials used for authentication
+         *      and authorization. The assumption we are making here is that the username is valid on the server,
+         *      the password is correct and the username has all the permissions necessary for performing the
+         *      subsequent operations.
+         */
+        ClientConfig clientConfig = ClientConfig.builder()
```

Comment:
Please, make it possible to pass via parameters some of these settings. You can have a look to existing samples for the parameter handling.

Reply:
As I mentioned in the PR description, I consciously avoided that. Here's why: 

> The samples are intended to be as simple as possible and easy for app developers to quickly observe and see security configuration in action, and therefore I've avoided taking any console inputs, encapsulating functionality into separate methods, reusing code across reader and writer apps (except for the Constants file) and so on. There are other examples in the project that demonstrate all of those aspects.

What do you say?

Reply:
I don't know if I agree on that. You are right in that samples should be as easy as possible to execute, and therefore when executing it "without parameters" (i.e., `bin/SecureReader` and `bin/SecureWriter`), the parameter values can be the ones defined in `Constants (i.e., we assume a standalone instance that runs locally). This is good, and most samples work like this. 

However, samples are also used by more advanced users (even ourselves) as a mean of rapidly testing the functionality of a deployment. So, imagine that we deploy a Pravega cluster with security enabled. It would be great to do some tests to verify that the security settings work correctly by using the samples. To this end, we would need to change the `--controller` parameter to the new deployment, and execute some operations against it. Doing this is not easy right now, as it requires to change the `Constants` file and recompile instead of using input parameters in the command line.

Reply:
As suggested, I've added support for executing the provided security examples via CLI.